### ISSUE TITLE
Low download bandwidth when using lgogdownloader

### ISSUE THREAD

**Issue by shmerl** [Sude-/lgogdownloader#117]
I have a gigabit connection, and when downloading using the browser from GOG, I'm easily hitting 90-100 MB/s. With lgogdownloader, I'm getting 2 MB/s, and it even often degrades to lower than 1.

Debian testing, lgogdownloader from the repo (3.2-1+b1).

**Comment by Sude-** [comment]
35192fd adds option to set user agent

**Comment by shmerl** [comment]
Heh, I was just in the middle of implementing a patch. Thanks!

**Comment by shmerl** [comment]
Well, spoofing user agent didn't help with this issue :(

**Comment by shmerl** [comment]
Yep, switching to building with `libcurl4-openssl-dev` seems to have some effect. So gnutls is causing it?

**Comment by YangtseSu** [comment]
The download speed is too slow, and often timeout.
But the browser or use aria2 is OK.
Can you add options to use download tools other than curl?

$ lgogdownloader --download-file factorio/en3installer0
Downloading: ./factorio_en_0_15_37_15778.sh
  0% ▕                          ▏ 0.00/0.00MB @ 0.00kB/s ETA: 0s
Retry 1/3 (timeout)
  0% ▕                          ▏ 0.00/0.00MB @ 0.00kB/s ETA: 0s
Retry 2/3 (timeout)
  0% ▕                          ▏ 0.00/0.00MB @ 0.00kB/s ETA: 0s
Retry 3/3 (timeout)
  0% ▕                          ▏ 0.00/0.00MB @ 0.00kB/s ETA: 0s

**Comment by YangtseSu** [comment]
--timeout=60 did not help.

**Comment by Sude-** [comment]
> The download speed is too slow, and often timeout.
> --timeout=60 did not help.

90dcaf0 adds options to set low speed timeout values for libcurl
Try different values for ```--lowspeed-rate``` and ```--lowspeed-timeout```

**Comment by YangtseSu** [comment]
The download can not start,The new options did no help.
The download speed is about 6MiB/s using aria2c.

**Comment by MafiaInc** [comment]
I'm also experiencing really low bandwidth ratio when downloading with LGOGDownloader 3.4. Currently downloading Heroes 3 Complete bin, around 954 MB with average speed of 400 kB/sec. My usual download speed is 70 Mbit/s, I'm on a wired, not Wi-Fi connection. I did test with curl itself (took URL via browser plugin CurlWget) and I downloaded the file with average speed of 7500 kB/sec. Let me know if I can help with more tests or information if you need it.

**Comment by ToddAWalter** [comment]
Did anyone ever confirm if GOG limits download rates?  I never get more than 6MB/s split across all threads.

**Comment by ToxicFrog** [comment]
No idea on the root cause, but whatever it is it's still an issue; browser and Galaxy downloads usually peak around 30-40MB/s, lgogdownloader on the same network is 2-3MB/s.
